cdap-docs/reference-manual/source/release-notes.rst

```diff
@@ -126,10 +126,20 @@ Known Issues
   the master log indicating that in-transit (emitted, but not yet processed) metrics
   system messages could not be decoded (*Failed to decode message to MetricsRecord*). This
   is because of a change in the format of emitted metrics, and can result in a small
-  amount of metrics data points being lost. (`CDAP-745
+  amount of metrics data points being lost (`CDAP-745
   <https://issues.cask.co/browse/CDAP-745>`__).
 - Writing to datasets through Hive is not supported in CDH4.x
   (`CDAP-988 <https://issues.cask.co/browse/CDAP-988>`__).
+- A race condition resulting in a deadlock can occur when a TwillRunnable container
+  shutdowns while it still has Zookeeper events to process. This occasionally surfaces when
+  running with OpenJDK or JDK7, though not with Oracle JDK6. It is caused by a change in the
+  ``ThreadPoolExecutor`` implementation between Oracle JDK6 and OpenJDK/JDK7. Until Twill is
+  updated in a future version of CDAP, a work-around is to kill the errant process. The Yarn
+  dashboard at port 8080 will provide the application id of the process; the command ``yarn
```

Comment:
Actually, one can also use the `yarn` command to list out all running applications by:

`yarn application -list -appStates RUNNING`

And to kill, should be `yarn application -kill <app_id>`. Sorry for the mis-info I gave earlier.

This is the complete help from the `yarn application` command

```
 -appStates <States>        Works with -list to filter applications based
                            on input comma-separated list of application
                            states. The valid application state can be one
                            of the following:
                            ALL,NEW,NEW_SAVING,SUBMITTED,ACCEPTED,RUNNING,
                            FINISHED,FAILED,KILLED
 -appTypes <Types>          Works with -list to filter applications based
                            on input comma-separated list of application
                            types.
 -help                      Displays help for all commands.
 -kill <Application ID>     Kills the application.
 -list                      List applications from the RM. Supports
                            optional use of -appTypes to filter
                            applications based on application type, and
                            -appStates to filter applications based on
                            application state
 -status <Application ID>   Prints the status of the application.
```